crates/rspack_plugin_devtool/src/lib.rs

```diff
@@ -570,6 +570,11 @@ async fn process_assets(&self, compilation: &mut Compilation) -> Result<()> {
     compilation.emit_asset(source_filename.to_owned(), source_asset.clone());
     if let Some((source_map_filename, source_map_asset)) = source_map {
       compilation.emit_asset(source_map_filename.to_owned(), source_map_asset.clone());
+      compilation.chunk_by_ukey.values_mut().for_each(|v| {
+        if v.files.contains(&source_filename) {
+          v.auxiliary_files.insert(source_map_filename.to_owned());
+        }
+      });
```

Comment:
Inserting `auxiliary_files` here is not a good choice. It is better to add them before the following code snippet:

```
mapped_asstes.push(MappedAsset {
    asset: (filename, asset.clone()),
    source_map: Some((source_map_filename, source_map_asset)),
});
```

Additionally, you should remove the following TODO comments:

```rust
// TODO
// chunk.auxiliary_files.add(filename);
```

Reply:
thx a lot for your suggestion, will look into it

Reply:
Hey @SyMind, Could you give me some guidance?

I checked the codes, and I found the borrowing rule in that method is not allowing me to change auxiliary_files, do you have any ideas? the only way I can see right now is to change `auxiliary_files` definition to `RefCell<HashSet<String>>`, but  it also will change a lot of places, not sure if it's the good direction

Reply:
I haven't considered it thoroughly. let's stick with your revisions.